CMakeLists.txt

```diff
@@ -79,6 +79,12 @@ if(NOT APPLE_IOS AND NOT ANDROID)
 	option(ENABLE_MONOLITHIC_INSTALL "Install everything in single directory on Linux and Mac" OFF)
 endif()
 
+# On Linux, use -DCMAKE_CXX_COMPILER_LAUNCHER=ccache instead.
+# The iOS and MSVC builds each require some configuration, which is enabled by the following option:
+if(APPLE_IOS OR MSVC)
```

Comment:
Xcode can also be used on macOS, that's why I suggested checking for the cmake generator that's used

Reply:
For the macOS builds, `-DCMAKE_CXX_COMPILER_LAUNCHER=ccache` is enough, though.

Reply:
with `-G Xcode`? are you sure? CI uses ninja.

Reply:
I mean `cmake -DCMAKE_CXX_COMPILER_LAUNCHER=ccache --preset macos-conan-ninja-release` and `cmake -DCMAKE_CXX_COMPILER_LAUNCHER=ccache --preset macos-arm-conan-ninja-release`

You can check the ccache stats in the CI log.

Reply:
I'm talking about dev workflow, not about CI. Example: I'm on macOS and run `cmake -G Xcode -DCMAKE_CXX_COMPILER_LAUNCHER=ccache <other params>` with intention to take advantage of ccache - it wouldn't work. If I am to check available options, `ENABLE_CCACHE` won't be listed for me.

Btw just noticed that the new logic is only inside iOS, instead it should be on the top level with cmake generator check.

Reply:
I've moved the XCode ccache config to the top level. Shouldn't the condition be
`if(ENABLE_CCACHE AND CMAKE_GENERATOR MATCHES "Xcode")`?

Maybe you could fork my repo and make the changes as well as test them on macOS.

Reply:
MATCHES isn't needed for Xcode, as the generator name is just `Xcode`

not sure why you changed MSVC to generator check though

Reply:
I have no experience with CMake generators.

AFAIK, the iOS and MSVC builds both needed more config than just CMAKE_CXX_COMPILER_LAUNCHER. So both would use the ENABLE_CCACHE option. But depending on the actual generator one uses, different configuration steps are taken.